base/ca/src/main/java/com/netscape/cms/profile/common/CAEnrollProfile.java

```diff
@@ -531,4 +532,113 @@ public void execute(Request request) throws EProfileException, ERejectException
         }
     }
 
+    private Map<String, byte[]> convertP12Password (Request request) throws EProfileException {
+        String method = "CAEnrollProfile: convertP12Password: ";
+        Map<String, byte[]> returnPass = null;
+
+        String p12passwd = request.getExtDataInString("serverSideKeygenP12Passwd");
+
+        org.mozilla.jss.crypto.X509Certificate transCert = null;
+
+        CAEngine engine = CAEngine.getInstance();
+        CertificateAuthority ca = engine.getCA();
+        CAConfig caConfig = ca.getConfigStore();
+        ConnectorsConfig connectorsConfig = caConfig.getConnectorsConfig();
+        ConnectorConfig kraConnectorConfig = connectorsConfig.getConnectorConfig("KRA");
+
+        try {
+            CryptoManager cm = CryptoManager.getInstance();
+            String transportNickname = kraConnectorConfig.getString("transportCertNickname", "KRA Transport Certificate");
+            transCert = cm.findCertByNickname(transportNickname);
+        } catch (Exception e) {
+            logger.debug(method + "'KRA transport certificate' not found in nssdb; need to be manually setup for Server-Side keygen enrollment");
+            throw new EProfileException(CMS.getUserMessage("CMS_MISSING_KRA_TRANSPORT_CERT_IN_CA_NSSDB"));
+
+            /* future; cert nickname can't be controlled yet at import in jss
+            logger.debug(method + "KRA transport certificate not found in nssdb; getting from CS.cfg");
+            transportCertStr = connectorsConfig.getString("KRA.transportCert", "");
+            logger.debug(method + "transportCert found in CS.cfg: " + transportCertStr);
+
+            byte[] transportCertB = Utils.base64decode(transportCertStr);
+            logger.debug(method + "transportCertB.length=" + transportCertB.length);
+            // hmmm, can't yet control the nickname
+            transCert = cm.importCACertPackage(transportCertB);
+            logger.debug(method + "KRA transport certificate imported");
+            */
+        }
+
+        try
+        {
+            // todo: make things configurable in CS.cfg or profile
+            CryptoToken ct =
+                CryptoUtil.getCryptoToken(CryptoUtil.INTERNAL_TOKEN_NAME);
+            if (ct == null)
+                logger.debug(method + "crypto token null");
```

Comment:
We probably should throw an exception if the token is not available. Alternatively, we probably can assume that the internal token is always available so no need to check for `null`.

Reply:
Actually, there is no action beside the log. If the token is null there will be exceptions later in the code.

I have investigated some other places where the token is retrieved and this is never verified. I am removing this line. If a log or exception is needed then it can be in the `getCryptoToken(<name>)` which log or raise an exception if it cannot be retrieved.